.github/workflows/sanitizer.yml

```diff
@@ -0,0 +1,48 @@
+name: Sanitize
+
+on: workflow_dispatch
+
+jobs:
+  cargo-careful:
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/osgeo/gdal:ubuntu-full-3.6.4
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install build deps
+        shell: bash
+        # Note: clang, etc. are needed so cargo-careful can build std with extra checks enabled
+        run: |
+          apt-get update -y
+          apt-get install build-essential pkg-config libclang-dev curl -y
+          curl --proto '=https' --tlsv1.2 --retry 10 --retry-connrefused -fsSL "https://sh.rustup.rs" | sh -s -- --profile minimal --default-toolchain none -y
+          echo "${HOME}/.cargo/bin" >> $GITHUB_PATH
+
+      - name: Setup building
+        run: |
+          export CC="clang-9"
+          export CXX="clang++-9"
+
+      - name: Install Rust nightly
+        run: |
+          rustup toolchain install nightly --profile minimal --component rust-src
+          rustup override set nightly
+
+      - name: Install cargo-careful
+        run: |
+          CAREFUL=${HOME}/.cargo/bin/cargo-careful
+          curl -Lo "${CAREFUL}" https://github.com/RalfJung/cargo-careful/releases/download/v0.3.4/cargo-careful.x86_64-unknown-linux-musl
```

Comment:
Maybe extract the version into a variable.

Reply:
Successful run post variable introduction:

https://github.com/metasim/gdal/actions/runs/5018034228/jobs/8996921323